sdk/nodejs/tests/runtime/langhost/cases/019.stack_exports/index.js

```diff
@@ -0,0 +1,21 @@
+function main() {
+    return new Promise((resolve) => {
+        resolve({
+            a: new Promise((resolve2) => {
+                resolve2({
+                    x: new Promise((resolve3) => {
+                        resolve3(99);
+                    }),
+                    y: "z",
+                });
+            }),
+            b: 42,
+            c: {
+                d: "a",
+                e: false,
+            },
+        });
+    });
+}
+
+module.exports = main();
```

Comment:
want to have an equivalent test that just uses an async function? (perhaps using TS as well)?

Reply:
Doesn't that compile down to the same thing?

Reply:
nope :)   and, it's unlikely there would be any issues found that way.  but i someone like it for completeness. i.e. i envision many people having an async entrypoint function that they'll just export.  Having an explicit test for that owuld be nice.

Note: i love this change.  It effectively solves the JS problem of "you can't await at the top level" for me.  The pattern you could now use is reasonably clear and easy and effective :)

Reply:
Oh you mean with newer Node runtimes. Don't we support downstream non-async capable runtimes? Also, we don't run any TypeScript in these tests, so we'd need to come up with some way of compiling them.......